pages/studentGradebook/studentGradebook.sql

```diff
@@ -0,0 +1,35 @@
+-- BLOCK select_assessment_instances
+SELECT
+    a.id AS assessment_id,
+    a.number AS assessment_number,
+    a.order_by AS assessment_order_by,
+    CASE
+        WHEN a.multiple_instance THEN a.title || ' instance #' || ai.number
+        ELSE a.title
+    END AS title,
+    aset.id AS assessment_set_id,
+    aset.abbreviation AS assessment_set_abbreviation,
+    aset.name AS assessment_set_name,
+    aset.heading AS assessment_set_heading,
+    aset.color AS assessment_set_color,
+    aset.number AS assessment_set_number,
+    CASE
+        WHEN a.multiple_instance THEN aset.abbreviation || a.number || '#' || ai.number
+        ELSE aset.abbreviation || a.number
+    END AS label,
+    ai.id AS assessment_instance_id,
+    ai.number AS assessment_instance_number,
+    ai.score_perc AS assessment_instance_score_perc,
+    ai.open AS assessment_instance_open,
+    (lag(assessment_set_id) OVER (PARTITION BY aset.id ORDER BY a.order_by, a.id, ai.number) IS NULL) AS start_new_set
+FROM
+    assessment_instances AS ai
+    JOIN assessments AS a ON (a.id = ai.assessment_id)
+    JOIN course_instances AS ci ON (ci.id = a.course_instance_id)
+    JOIN assessment_sets AS aset ON (aset.id = a.assessment_set_id)
+WHERE
+    ci.id = $course_instance_id
```

Comment:
Mirror @echuber2's concern on slack, maybe perform an access rule check here to see if the assignment was available to the student at one point?

Reply:
For an `assessment_instance` to be created, the student would have needed to have accessed the assessment at one time. What Matt meant is this page isn't checking for a current access rule to allow viewing because it's only showing the overall score.